src/ziggurat/config.clj

```diff
@@ -108,3 +110,90 @@
 (defn -get [^String key]
   (let [config-vals (get config (keyword key))]
     (java-response config-vals)))
+
+(def ^:private consumer-config-mapping-table
+  {:auto-offset-reset-config        :auto-offset-reset
+   :commit-interval-ms              :auto-commit-interval-ms
+   :consumer-group-id               :group-id
+   :default-api-timeout-ms-config   :default-api-timeout-ms
+   :key-deserializer-class-config   :key-deserializer
+   :session-timeout-ms-config       :session-timeout-ms
+   :value-deserializer-class-config :value-deserializer})
+
+(def ^:private producer-config-mapping-table
+  {:key-serializer-class   :key-serializer
+   :retries-config         :retries
+   :value-serializer-class :value-serializer})
+
+(def ^:private streams-config-mapping-table
+  {:auto-offset-reset-config           :auto-offset-reset
+   :default-api-timeout-ms-config      :default-api-timeout-ms
+   :changelog-topic-replication-factor :replication-factor
+   :session-timeout-ms-config          :session-timeout-ms
+   :stream-threads-count               :num-stream-threads})
+
+(def config-mapping-table
+  (merge consumer-config-mapping-table
+         producer-config-mapping-table
+         streams-config-mapping-table))
+
+(def ^:private non-kafka-config-keys
+  [:channels
+   :consumer-type
+   :input-topics
+   :join-cfg
+   :oldest-processed-message-in-s
+   :origin-topic
+   :poll-timeout-ms-config
+   :producer
+   :thread-count])
+
+(defn- to-list
+  [s]
+  (if (empty? s)
+    (list)
+    (list s)))
+
+(defn- to-string-key
+  [mt k]
```

Comment:
we can rename mt to something better. 

For a few minutes, I couldn't understand that mt is mapping-table. I think renaming this will make more readable for others who are not actively working on this code.

Reply:
done